EndGameConditions

Jason0303132 opened the pull request:
Added EndGameConditons_UseCase package, which contains EndGameConditions (Semi-implemented, with basic components done) and an empty EndGameConditions_Controller, (Also deleted tutorial file)

Jason0303132 committed:
Implemented endGame checker, checkWInMoney, checkWinSpecies, and checkLose

Note for endGame method, when it is called by nextDay, it will return "game not over" if the win/lose conditions have not been met, so nextDay can use a checker to determine whether the game can continue.

Jason0303132 committed:
Added EndGameConditons_UseCase package, which contains EndGameConditions (Semi-implemented, with basic components done) and an empty EndGameConditions_Controller, (Also deleted tutorial file)

Jason0303132 committed:
Merge branch 'main' into 1-feature-3-winninglosing-conditions

Jason0303132 committed:
Added EndGameConditons_UseCase package, which contains EndGameConditions (Semi-implemented, with basic components done) and an empty EndGameConditions_Controller, (Also deleted tutorial file)

dianzrong reviewed (commented):
Why does EndgameConditions extend InventoryManager? It might be a better idea to just call InventoryManager

Jason0303132 committed:
Added EndGameConditons_UseCase package, which contains EndGameConditions (Semi-implemented, with basic components done) and an empty EndGameConditions_Controller, (Also deleted tutorial file)

With changes so that EndgameCondtions no longer is a child of Inventory Manager.

DavidGorospe reviewed (approved):
- Good package organization by making the EndGameConditions use case its own package
- Methods accurately check each win/lose condition and display the correct message
- EndgameConditions class correctly accesses the UseCase Interactors

dianzrong reviewed (approved):
I think the code works and is clean. One thing I'm wondering is should we redo some of the return strings? 
- Maybe it would be better to separate the string into 2 sentences with the requirement and what they currently have. Ex: "You have acquired all of the different species of crops and animals required to win. Currently you have __ species"
- Also maybe it would be better to remove "amount" at the end because I think it would make more sense then
